### PR TITLE
Containerizing schematic (and update dependencies as needed)

### PR DESCRIPTION
**Pull request by BrunoGrandePhD**
You can now run schematic using [Docker](https://hub.docker.com/r/sagebionetworks/schematic/tags)! My goal is to make it easier for schematic users and developers to debug issues by ensuring specific versions are being used. The container can be further optimized, but I'm hoping this can serve as a starting point. 

**Edit:** I think the only missing part for getting the tests to work is creating a `TestDataset3.10` folder on [Synapse](https://www.synapse.org/#!Synapse:syn24239829). Is there documentation on how to create these folders? 

```console
$ docker run sagebionetworks/schematic --help

Usage: schematic [OPTIONS] COMMAND [ARGS]...

  Command line interface to the `schematic` backend services.

Options:
  -v, --verbosity LVL  Either CRITICAL, ERROR, WARNING, INFO or DEBUG
  -h, --help           Show this message and exit.

Commands:
  init      Initialize mode of authentication for schematic.
  manifest  Sub-commands with Manifest Generation utilities/methods.
  model     Sub-commands for Metadata Model related utilities/methods.
  schema    Sub-commands for Schema related utilities/methods.
```

I'm also including a GitHub Actions CI workflow that will auto-publish containers to Docker Hub. I created the `sagebionetworks/schematic` repository on Docker Hub, and I've already configured the auth token as a secret on this repo. I'll need to add a few people to the `schematic@sagebase.org` Google Group I created to act as the email of the `schematicbot` Docker service account being used to upload to Docker Hub. 

Note that this GitHub Actions workflow will publish a tag on every release, updating the `latest` tag in the process. There's also the option to perform a manual release (_i.e._ `workflow_dispatch`), which will create a tag called `commit-03002fc`, where the second bit is the short commit SHA ([example](https://hub.docker.com/r/sagebionetworks/sagetasks/tags) on Docker Hub). This can be useful when you want to distribute a potential fix with users before issuing an official tagged release. I've included a screenshot below of the manual trigger from another repo with a similar setup. 

Lastly, I had a lot of trouble getting `poetry install` to work in the container, so I had to fiddle with the dependencies a bit. Let me know if any of these changes are problematic. 

## Screenshot of Workflow Dispatch

image

### CODE REVIEW

**Comment by GiaJordan** [comment]
> Edit: I think the only missing part for getting the tests to work is creating a TestDataset3.10 folder on [Synapse](https://www.synapse.org/#!Synapse:syn24239829). Is there documentation on how to create these folders?

@BrunoGrandePhD currently no, that was added a couple days ago. For now, I can create the folder, and update the test that's currently failing to use version `3.10` as well. And then add docs for how to update in the future!

**Comment by BrunoGrandePhD** [comment]
Thanks, @GiaJordan! I was able to pass all of the tests with the new test dataset folder!

**Comment by GiaJordan** [comment]
@BrunoGrandePhD Good to hear!
I was also able to implement the library @linglp found in PR #880, and that should simplify the dummy data that's necessary for the tests and eliminate the need for a version specific dataset in the future

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# ignore .git and .cache folders
+.git
+.cache
```
Comment (BrunoGrandePhD) [inline .dockerignore:3]: List Synapse config

**Comment by BrunoGrandePhD** [comment]
I was able to reduce the test redundancy as per my comment [here](https://github.com/Sage-Bionetworks/schematic/pull/880#pullrequestreview-1087550335). 

This PR is ready for review!

**Review by GiaJordan — APPROVED** [review]
Tentatively approving after review at Code Space on 8/26, but also requesting review from @afwillia because of his previous involvement in dockerizing schematic. Thanks @BrunoGrandePhD!

**Comment by afwillia** [comment]
@BrunoGrandePhD thank you this is fantastic! I was able to pull the image and validate a manifest from the command line.

**Review by afwillia — APPROVED** [review]

**Comment by milen-sage** [comment]
🚀

**Pull request merged**